integration-cli/docker_api_containers_unix_test.go

```diff
@@ -0,0 +1,49 @@
+// +build !windows
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"syscall"
+
+	"github.com/docker/docker/integration-cli/checker"
+	"github.com/docker/docker/integration-cli/request"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/go-check/check"
+)
+
+func (s *DockerSuite) TestContainerAPIArchiveWithHardlink(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	containerID := makeTestContainer(c, testContainerOptions{addContent: true})
+
+	tmpDir := getTestDir(c, "test-cp-from-dir-with-hardlink")
+	defer os.RemoveAll(tmpDir)
+
+	srcPath := "/dir6"
+	dstPath := cpPath(tmpDir, "dir6_")
+	dstFile := cpPath(tmpDir, "dir6_/hardLinkFile1")
+	endpoint := fmt.Sprintf("/containers/%s/archive?path=%s", containerID, srcPath)
+	res, body, err := request.SockRequestRaw("GET", endpoint, nil, "application/x-tar", daemonHost())
+	c.Assert(err, checker.IsNil)
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
+
+	dstDir, dstBase := archive.SplitPathDirEntry(dstPath)
+	_, srcBase := archive.SplitPathDirEntry(srcPath)
+	copyArchive := archive.RebaseArchiveEntries(body, srcBase, dstBase)
```

Comment:
Sorry, I wasn't clear. I meant to ask, why is `RebaseArchiveEntries()` being called here?

It seems to just rename files inside the archive, but I don't understand why that's necessary for this test. We should be just asserting that the archive contains the files we expect.

I'd like to suggest another approach. Instead of making this test about just hardlinks, we should make this a general purpose test for the archive endpoint.

It looks like we have a good way of creating a container with a bunch of different types of files (`makeTestContainer()`), so all we need is to assert that all the files in the archive are as we expect and we can test everything with this one case.

Does that make sense?